druid-shell/src/platform/gtk/window.rs

```diff
@@ -624,23 +624,26 @@ impl<'a> WinCtx<'a> for WinCtxImpl<'a> {
     }
 
     fn request_timer(&mut self, deadline: std::time::Instant) -> TimerToken {
-        let interval = time_interval_from_deadline(deadline);
-        let token = next_timer_id();
+        let interval = deadline
+            .checked_duration_since(std::time::Instant::now())
+            .unwrap_or_default()
+            .as_millis() as u32;
```

Comment:
This will overflow when the deadline is more than 48 days into the future or whatnot. I don't think this is important to solve right now, but maybe a note.

Reply:
fair, I can be more defensive here.